### PR TITLE
migrate to ZIO 2

### PR DESCRIPTION
**Pull request by alterationx10**
To address #36 , I followed  the [Migration Guide](https://zio.dev/guides/migrate/zio-2.x-migration-guide), and it seems like it all compiles 😄

I also bumped the sbt version, as the one listed doesn't seem to play well with jdk 18

### CODE REVIEW

File: zio-crypto/shared/src/main/scala/zio/crypto/random/SecureRandom.scala
Comment (alterationx10) [inline zio-crypto/shared/src/main/scala/zio/crypto/random/SecureRandom.scala:41]: Is this the appropriate use of `Scoped`? Wasn't sure for this case, but seems ok.

**Comment by alterationx10** [comment]
I guess there doesn't seem to be any CI tests run? "Works on my machine" if it makes anyone feel better 🤣 

```
[info] 96 tests passed. 0 tests failed. 0 tests ignored.
```

**Comment by vigoo** [comment]
> I guess there doesn't seem to be any CI tests run? "Works on my machine" if it makes anyone feel better 🤣
> 
> ```
> [info] 96 tests passed. 0 tests failed. 0 tests ignored.
> ```

As first contributor we have to explicitly approve CI runs for you, after this is merged it will no longer be necessary :)

File: zio-crypto/shared/src/main/scala/zio/crypto/encryption/HybridEncryption.scala
Comment (vigoo) [inline zio-crypto/shared/src/main/scala/zio/crypto/encryption/HybridEncryption.scala:83]: This could be `ZIO.serviceWithZIO(_.encrypt(plainText, key))`

File: zio-crypto/shared/src/main/scala/zio/crypto/random/SecureRandom.scala
Comment (vigoo) [reply zio-crypto/shared/src/main/scala/zio/crypto/random/SecureRandom.scala:41]: No this will release the scope at the end of the creation of the layer and leak out the released fiber ref.

You need to use `ZLayer.scoped { ... }` which creates a `Scope` for the lifecycle of the layer

File: zio-crypto/shared/src/test/scala/zio/crypto/encryption/HybridEncryptionSpec.scala
Comment (vigoo) [inline zio-crypto/shared/src/test/scala/zio/crypto/encryption/HybridEncryptionSpec.scala:19]: I think `a != b` is easier to read :)

File: zio-crypto/shared/src/main/scala/zio/crypto/random/SecureRandom.scala
Comment (alterationx10) [reply zio-crypto/shared/src/main/scala/zio/crypto/random/SecureRandom.scala:41]: 🤦 Oh yeah. I'll get this updated!

File: zio-crypto/shared/src/main/scala/zio/crypto/encryption/HybridEncryption.scala
Comment (alterationx10) [reply zio-crypto/shared/src/main/scala/zio/crypto/encryption/HybridEncryption.scala:83]: The scalafix from the migration did all of these. I noticed it changed `.accessM` to `environmentWithZIO`, but it didn't updated anything that was `.access`. I noticed this behavior with a work project I was testing migrations on for an earlier RC. Is this something that might need to be updated in the scalafix rule?

Either way, it probably makes more sense to make them `. serviceWithZIO` - I'll update them.

File: zio-crypto/shared/src/test/scala/zio/crypto/encryption/HybridEncryptionSpec.scala
Comment (alterationx10) [reply zio-crypto/shared/src/test/scala/zio/crypto/encryption/HybridEncryptionSpec.scala:19]: 💯 I agree. I was mostly just following the auto-suggestion from the zio plugin while I was fixing these specs. I'll update these too.

File: zio-crypto/shared/src/main/scala/zio/crypto/encryption/HybridEncryption.scala
Comment (alterationx10) [reply zio-crypto/shared/src/main/scala/zio/crypto/encryption/HybridEncryption.scala:83]: 🤦 Didn't notice the `.get` right away.

**Comment by alterationx10** [comment]
I made a slight change to the ci.yml file

ZIO 2 was (cross) built against 3.1.2, and this is using (incompatible) 3.0.1.

I updated the Scala matrix to match that of the current core ZIO library.

The version bump to Scala 2.13.8  meant updating a couple dependencies as well:

```
SilencerVersion => 1.7.8 (matching core zio lib)
sbt-scalafix => 0.10.0 (mathcing core zio lib)
kind-projector => 0.13.2 (bumped to latest)
```

**Review by vigoo — APPROVED** [review]

**Pull request merged**
